test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateSaveResultTests.cs

```diff
@@ -0,0 +1,487 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="BulkUpdateSaveResultTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.OData.Client.TDDUnitTests.Tests
+{
+    /// <summary>
+    /// Unit tests for the BulkUpdateSaveResult class.
+    /// </summary>
+    public class BulkUpdateSaveResultTests
+    {
+        private readonly DataServiceContext context;
+        private readonly RequestInfo requestInfo;
+        private readonly Serializer serializer;
+        private readonly HeaderCollection headers;
+        private BulkUpdateGraph bulkUpdateGraph;
+
+        public BulkUpdateSaveResultTests()
+        {
+            this.context = new DataServiceContext(new Uri("http://www.odata.org/service.svc")).ReConfigureForNetworkLoadingTests();
+            this.requestInfo = new RequestInfo(context);
+            this.serializer = new Serializer(this.requestInfo);
+            this.headers = new HeaderCollection();
+            bulkUpdateGraph = new BulkUpdateGraph(this.requestInfo);
+        }
+
+        [Fact]
+        public void SerializeADeltaEntry_With_TwoNavigationLinks()
+        {
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+
+            this.context.AttachTo("Persons", person);
+            this.context.AttachTo("Cars", car1);
+            this.context.AttachTo("Cars", car2);
+            this.context.AddLink(person, "Cars", car1);
+            this.context.AddLink(person, "Cars", car2);
+
+            this.bulkUpdateGraph = this.GetBulkUpdateGraph(person);

```

Comment:
I think it would be great to have tests that actually use the `DeepUpdate` method. The current tests you have rely heavily on internal implementation details and are not sufficient to give confidence that `DeepUpdate`, which is what the user will interact with, actually works as expected.

Reply:
Yes..I think tests for `DeepUpdate` will be added when we add the deserialization changes.. These tests are centered more on serialization.